src/DistributedLock.AzureStorage/AzureStorageDistributedLock.cs

```diff
@@ -149,29 +147,37 @@ public async Task ReleaseLockAsync()
         /// <param name="key">key for the lock.</param>
         /// <param name="optionsBuilder">options to build the locker.</param>
         /// <returns></returns>
-        public static async Task<IDistributedLock> CreateAsync(string key, Action<AzureStorageDistributedLockOptions> optionsBuilder = null)
+        public static IDistributedLock Create(string key, Action<AzureStorageDistributedLockOptions> optionsBuilder = null)
         {
             var options = new AzureStorageDistributedLockOptions(key);
             optionsBuilder?.Invoke(options);
 
             var locker = new AzureStorageDistributedLock(options);
 
-            await locker.Init().ConfigureAwait(false);
             return locker;
         }
 
+        /// <summary>
+        ///     Create an AzureStorage Distributed Locker. 
+        /// </summary>
+        /// <param name="key">key for the lock.</param>
+        /// <param name="optionsBuilder">options to build the locker.</param>
+        /// <returns></returns>
+        public async static Task<IDistributedLock> CreateAsync(string key, Action<AzureStorageDistributedLockOptions> optionsBuilder = null)
```

Comment:
@Master-Oliveira this async await can be avoided right ? to avoid an extra state machine

Reply:
Completely right, fixed.